### DISCUSSION TITLE
How do I control the fan on my framework?

### DISCUSSION THREAD

**Discussion by LovelyA72** [DHowett/FrameworkWindowsUtils#2] [Q&A] [answered]
I read that there is `fanduty` and `autofanctrl` command that I can use to control my fan speed. How can I do that on Windows?

**Comment by DHowett** [comment, marked as answer]
Right now there's no easy way to do this.

With the test tool from this distribution, `FauxECTool`, you can sort of accomplish it with raw commands (as fauxectool doesn't support symbolic commands 😄)

### Set fan 0 duty% to 0x00000064, or 100%
```
fauxectool 0x24 d64
```

`0x24` is `EC_CMD_PWM_SET_FAN_DUTY`

> **Note**: there is no way to specify the percentage in decimal

### Set fan 0 RPM target to 4000 (0x00000FA0) directly
```
fauxectool 0x21 dFA0
```

`0x21` is `EC_CMD_PWM_SET_FAN_TARGET_RPM`

> **Note**: there is no way to specify the RPM in decimal

> **Note**: the fans on this board operate up to 6800 RPM

### Restore fan 0x00 to automatic thermal control
```
fauxectool 0x52 b0
```

`0x52` is `EC_CMD_THERMAL_AUTO_FAN_CTRL`

I've been testing a build of `ectool` (the Linux version) that works on Windows, which should make this much easier. It is not yet ready for release.

**Comment by DHowett** [comment]
I've got good news! I brought the upstream `ectool` code out into a [separate repository](https://gitlab.howett.net/DHowett/ectool) and ported it to Windows.

Builds are available over there. Just check out the latest `windows/x64` artifacts:
![image](https://user-images.githubusercontent.com/189190/196216082-ea2ff91e-fc37-4566-a50a-beb6445707e4.png)
